addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1661,9 +1663,12 @@ const Wysiwyg = Widget.extend({
             e.preventDefault();
             const selection = this.odooEditor.document.getSelection();
             const containerSelector = '#wrap>*, [contenteditable], .oe_structure>*';
+            const blockEl = closestBlock(selection.anchorNode);
+            const closestDiv = closestElement(selection.anchorNode, 'div');
+            const isDocumentEmpty = closestDiv.childElementCount <= 1 && !(blockEl.previousSibling || blockEl.nextSibling) && !selection.anchorNode.textContent;
             let $deepestParent =
-                selection ?
-                    $(selection.anchorNode).parentsUntil(containerSelector).last() :
+                selection && !isDocumentEmpty ?
+                    $(lastLeaf(blockEl)).parentsUntil(containerSelector).last() :
```

Comment:
I don't understand. What happens with the previous code ? What does `$(selection.anchorNode).parentsUntil(containerSelector).last()` used to return in that case before your changes ?

Reply:
@dmo-odoo the previous code we will get jQuery object when the selection was empty and a jQuery object containing a p element when content was present.

The previous code would encounter this issue because the code contained a condition that added the current node until it found an element node from our anchor node. now when we do ctrl+a in empty selection we will get the p element as a anchor node, which was already an element node, and this would lead to an empty jQuery object being returned.

Reply:
@dhba-odoo Thanks for the explanation. Based on that I tried another fix. However, my fix changes the behavior of the select all a little bit so I need to ask a question to @sebgeelen to be sure.

I replaced the `parentsUntil` with `closestElement` (so that `selection.anchorNode` is also matched against the selector) but it has the opposite meaning sort of, that is it is going to stop at the first matched node whereas @sebgeelen's version with `parentsUntil` specifically targeted the upmost ancestor instead.

I had a try in different scenarios but couldn't find any place where the old behavior was actually better than the new one. Basically, from what I saw it is the same everywhere except in links, where my version selects the entire link whereas the current 15.0 code wouldn't select anything (due to the bug I'm fixing). I'm looking for a case where you might remember we specifically needed to target the highest parent in the tree. Do you remember such a case ?